### DISCUSSION TITLE
Cannot update service in DB-less mode

### DISCUSSION THREAD

**Discussion by black-snow** [Kong/kong#7995] [Help] [answered]
I found the related issue https://github.com/Kong/kong/issues/4373, that had no solution.

I'm just getting started with kong and I opted for the DB-less mode since it's less hassle / less to set up. So I set up kong OSS 2.6 and decK 1.8.2 (becaue why wouldn't you wanna be declarative?), wrote an example `kong.yaml` and tried to `deck sync` it:

> creating service xxx-service
Summary:
  Created: 0
  Updated: 0
  Deleted: 0
Error: 1 errors occurred:
        while processing event: {Create} service xxx-service failed: HTTP status 405 (message: "cannot create or update 'services' entities when not using a database")

Why is that? I read that you started working on the delcarative stuff in '19 so I figure this ain't no bleeding edge feature ... or is it? I can just restart the container with the new YAML but I cannot update it via the API (read decK)? :|

**Comment by kikito** [comment, marked as answer]
Hi, you are partially correct. It's true that restarting the container with a new YAML file is certainly one option, but it's not what we would recommend, since that would mean interrupting the service.

Most of the Admin API is read-only on DB-less mode, and will give you the 405 error that you just received if you attempt to modify individual entities. You might have missed that they are marked as `Not available in DB-less mode` in our docs, for example here're the docs for [Update Service](https://docs.konghq.com/gateway-oss/2.6.x/admin-api/#update-service)

However, the  `/config` Admin API endpoint is available on DB-less. It will accept the YAML file and will replace all the entities in one go. It does not perform a server restart.

You may find more information about `/config` on the [DB-less and Declarative Section](https://docs.konghq.com/gateway-oss/2.6.x/db-less-and-declarative-config/#loading-the-declarative-configuration-file) of our docs, and the list of everything that can be modified by The Admin API on DB-less mode [at the beginning of our Admin API docs](https://docs.konghq.com/gateway-oss/2.6.x/admin-api/#db-less-mode). Finally, all the methods on that page are marked with either a green "DB-less compatible" tag or a "Not compatible with DB-less" note.

**Reply by black-snow** [reply]
Thanks for the quick reply!

Alright, but shouldn't decK be agnostic to what mode is run? At the user side, I mean. If there's an endpoint to paste a new YAML to, decK could just do that. I figure that with the "normal" mode, decK does some shenanigans to only actually do the delta but it could probably detect that Kong runs in DB-less mode and just paste the whole thing, right?
To me, as a new user, this comes a bit unexpected. DB-less sounded way easier to set up and maintain so I opted for that for my playground. decK sounded like a perfect fit. But apparently they don't go together atm. :|

P.S.: btw, did you think about providing a terraform provider/ modules? It's configuration but I somehow feel that this would fit nicely into my inventory ;)
P.P.S.: As always, I might just be too stupid to read the docs :)

Cheers
